src/cipher.rs

```diff
@@ -23,21 +23,22 @@ use crate::{
 use rustls_result::{AlreadyUsed, NullParameter};
 
 /// An X.509 certificate, as used in rustls.
-/// Corresponds to `Certificate` in the Rust API.
-/// <https://docs.rs/rustls/latest/rustls/struct.Certificate.html>
-pub struct rustls_certificate {
+/// Corresponds to `CertificateDer` in the Rust pki-types API.
+/// <https://docs.rs/rustls-pki-types/latest/rustls_pki_types/struct.CertificateDer.html>
+pub struct rustls_certificate<'a> {
     // We use the opaque struct pattern to tell C about our types without
     // telling them what's inside.
     // https://doc.rust-lang.org/nomicon/ffi.html#representing-opaque-structs
     _private: [u8; 0],
+    _marker: PhantomData<&'a ()>,
```

Comment:
Calling this out as a more interesting change:

The `RustType` of the `rustls_certificate` type's `Castable` impl gains a lifetime as of the switch to `CertificateDer`, which means that the `rustls_certificate` type needs a lifetime too. 

That in turn requires adding a `PhantomData` marker since otherwise it would be an unbounded lifetime not referenced by a field. Based on [the FFI guide section on opaque structs](https://doc.rust-lang.org/nomicon/ffi.html#representing-opaque-structs) I think adding the phantom data reference is the correct solution here.